fsharp-backend/src/LibBackend/StaticAssets.fs

```diff
@@ -107,8 +97,115 @@ let latestDeployHash (canvasID : CanvasID) : Task<Option<string>> =
   |> Sql.parameters [ "canvasID", Sql.uuid canvasID; "branch", Sql.string branch ]
   |> Sql.executeRowOptionAsync (fun read -> read.string "deploy_hash")
 
+let allDeploysInCanvas
+  (canvasName : CanvasName.T)
+  (canvasID : CanvasID)
+  : Task<List<StaticDeploy>> =
+  Sql.query
+    "SELECT deploy_hash, created_at, live_at
+     FROM static_asset_deploys
+     WHERE canvas_id=@canvasID
+     ORDER BY created_at
+     DESC LIMIT 25"
+  |> Sql.parameters [ "canvasID", Sql.uuid canvasID ]
+  |> Sql.executeAsync (fun read ->
+    let deployHash = read.string "deploy_hash"
+
+    let status, lastUpdate =
+      match read.instantOrNone "live_at" with
+      | Some datetime -> Deployed, datetime
+      | None -> Deploying, read.instant "created_at"
+
+    { deployHash = deployHash
+      url = url canvasName deployHash Short
+      status = status
+      lastUpdate = lastUpdate })
+
+/// Creates a record of a new static asset deploy, returning a deployHash to be
+/// used by the actual upload process.
+let startStaticAssetDeploy
+  (user : Account.UserInfo)
+  (canvasID : CanvasID)
+  (canvasName : CanvasName.T)
```

Comment:
I believe we're not using canvasName here anymore?

Reply:
thanks